### PR TITLE
Eclipse Project import - Windows part

### PR DESCRIPTION
**Pull request by fape**
Based of #2046, Thanks to @rohankuruvilla
- Add some missing dependency
- Update READ.md-s with windows/cygwin specific part

### CODE REVIEW

File: samples/Cpp/TestCpp/proj.android/.cproject [hunk 1]
```diff
@@ -20,7 +18,7 @@
...
 							<targetPlatform id="org.eclipse.cdt.build.core.prefbase.toolchain.294545719.993860072" name=""/>
-							<builder arguments="${ProjDirPath}/build_native.sh" buildPath="${ProjDirPath}" command="bash" enabledIncrementalBuild="true" errorParsers="org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.CWDLocator" id="org.eclipse.cdt.build.core.settings.default.builder.1648661634" incrementalBuildTarget="" keepEnvironmentInBuildfile="false" managedBuildOn="false" name="Gnu Make Builder" parallelBuildOn="false" stopOnErr="true" superClass="org.eclipse.cdt.build.core.settings.default.builder">
+							<builder arguments="-o igncr ${ProjDirPath}/build_native.sh" buildPath="${ProjDirPath}" command="bash" enabledIncrementalBuild="true" errorParsers="org.eclipse.cdt.core.GmakeErrorParser;org.eclipse.cdt.core.CWDLocator" id="org.eclipse.cdt.build.core.settings.default.builder.1648661634" incrementalBuildTarget="" keepEnvironmentInBuildfile="false" managedBuildOn="false" name="Gnu Make Builder" parallelBuildOn="false" stopOnErr="true" superClass="org.eclipse.cdt.build.core.settings.default.builder">
```
Comment (minggo) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: What's the effect of "-o igncr"?

**Comment by minggo** [comment]
@rohankuruvilla
Could you help to review the pull request?
And i met a problem after merging your PR. The .project file is changed when i imported a sample project.

@fape, have your met the problem?

**Comment by minggo** [comment]
It is merged in this [PR](https://github.com/cocos2d/cocos2d-x/pull/2187)

File: samples/Cpp/TestCpp/proj.android/.cproject (hunk 1 above)
Comment (minggo) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: bash -o igncr /Users/minggo/SourceCode/cocos2d-x/samples/Cpp/HelloCpp/proj.android/build_native.sh 
bash: line 0: bash: igncr: invalid option name

The erro message when running on mac.

Comment (fape) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: "-o igncr" means "Ignore carriage return". In windows git uses windowsline ending(\r\n) to "checkout" files. But in cygwin linux lineending (\n) is the default. So Cygwins's bash try to interpret carriage return (\r) as a command. I'm going to find an other solution for this problem.

Comment (dumganhar) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: @fape 
You could disable `autocrlf` flag to false. In that case, git on windows will not modify the return flag to windows when you pull.

Comment (dumganhar) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: You could use `git config --global core.autocrlf false` to disable it.

Comment (minggo) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: Thank you @fape.

Now Eclipse will compile all projects when i open Eclipse.
Then i can not do anything with Eclipse.

Is there a solution to resolve it? Only compile a project when i click "build" option.

Comment (fape) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: @dumganhar , Thanks, I known this command. But I'd like to serve solution for default config, if it possible.

@minggo: 
- the "bash: line 0: bash: igncr: invalid option name" error message was disappeared on Mac?
- You can run app on your phone ("run as" > "android application" > choose you phone) and see "realtime" log cat output in eclipse (Window > show view > logcat). Unfortunately the android emulator does not support gles2 :S

Comment (minggo) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: @fape 
- I delete "-o igncr" now until we find better solution
- What i means is that, I don't want to Eclipse automatically build all projects(it is a long time) when i open it.

Comment (fape) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: @minggo 
- I found a better solution: In eclipse add new C/C++ build Environment variable **SHELLOPTS** with value `igncr`. I will update the readme files and delete `-o igncr` from build command in an other PR.
- Sorry. I am also noticed that. Try to manualy kill/stop the "Android Library Update" in progress view.

Comment (minggo) [inline samples/Cpp/TestCpp/proj.android/.cproject:21]: Thank you @fape .

**Comment by DavidRO87** [comment]
I've used the official, I tried it beacause I'm newer in cocos and I have some errors on Eclipse... but I still have the same pronlems

**Comment by fape** [comment]
As I see CocosPlayer was (re)moved in ec63e44026b99fab9c28ee6960508c78aaa6d019 commit.
But if you still have errors, I think the official forum ( http://www.cocos2d-x.org/forums ) is better place to ask. There are lot of people who can help.

**Comment by DavidRO87** [comment]
Thank you fape. :)

**Pull request closed without merging**